ospfd/ospf_vty.c

```diff
@@ -8926,7 +8926,7 @@ DEFUN (ip_ospf_area,
 	argv_find(argv, argc, "area", &idx);
 	areaid = argv[idx + 1]->arg;
 
-	if (ifp->vrf_id && !instance)
```

Comment:
This won't help. We'll still use `ospf_lookup_instance` for the default VRF lookup.
The correct thing to do is to compare `ifp->vrf_id` with `VRF_UNKNOWN` instead of zero (because zero is VRF_DEFAULT and the current check doesn't make sense at all).

P.S. This code doesn't work completely when the VRF does not exist yet. But fixing it would involve a lot more work and can be done separately.

Reply:
Well, it failed my simple test before my change and works after, so saying it won't help is certainly wrong.  If you don't have instance value, it relies on the vrf_id (which can be 0 for default) to find the process to use.  Are you saying that the lookup by vrf_id  logic doesn't work?  I do recognize I didn't try the unknown vrf_id case so that is work to be done.  Please explain a bit more about why what I've done won't work.

Reply:
Your code doesn't change the processing of the command in the default VRF at all.
The interface in the default VRF has vrf_id == 0. So your removal of `!instance` doesn't change that for the default VRF we call `ospf_lookup_instance`.

And calling `ospf_lookup_instance` is completely incorrect if you don't use instances, because all ospf structures in all VRF have `ospf->instance == 0` and `ospf_lookup_instance` just always returns the first entry in the list.
ospf struct for the default VRF just happened to be the first one in the list in your test, that's why it worked for you.

Reply:
As I said, if you change the condition to `ifp->vrf_id != VRF_UNKNOWN`, then we'll correctly call `ospf_lookup_by_vrf_id` for the default VRF.

Reply:
This makes much more sense :) Thanks!